in_progress/hd_shank_preprocessing.py

```diff
@@ -0,0 +1,151 @@
+from pathlib import Path
+import numpy as np
+from pydantic import BaseModel, Field
+from pyrsistent import v
+import spikeinterface as si
+import spikeinterface.preprocessing as spre
+
+
+########################################################################################################################
+# Preprocessing parameters
+
+class PhaseShiftParameters(BaseModel):
```

Comment:
> Last comment really! Should these be `@dataclass`?

Feel free to make as many comments as you like! :)

I used pydantic because I wanted to encode the descriptions in a way that could be parsed out elsewhere.

I am looping @luiztauffer in.

Reply:
Oups I didn't notic it was a pydantic model! Nevermind, looks great :) Merging!